packages/driver/src/cypress/error_messages.ts

```diff
@@ -1880,26 +1876,49 @@ export default {
 
         Cypress only considers the \`window\`, \`document\`, or any \`element\` to be valid DOM objects.`
     },
-    not_attached (obj) {
+    detached_during_actionability (obj) {
       return {
         message: stripIndent`\
-          ${cmd(obj.cmd)} failed because this element is detached from the DOM.
+          ${cmd(obj.name)} failed because the page updated while this command was executing. Cypress tried to locate elements based on this query:
 
-          \`${obj.node}\`
+          > ${subjectChainToString(obj.subjectChain)}
+
+          We initially found matching element(s), but while waiting for them to become actionable, they disappeared from the page. Common situations why this happens:
+            - Your JS framework re-rendered asynchronously
+            - Your app code reacted to an event firing and removed the element
 
-          Cypress requires elements be attached in the DOM to interact with them.
+          You can typically solve this by breaking up a chain. For example, rewrite:
 
-          The previous command that ran was:
+          > \`cy.get('button').click().click()\`
```

Comment:
I think this example is another area I would have expected to work under the hood like:
```
1. get button
2. click button
3. Try click button // DETACHED!! // get button // click button (if exists)
```
I would have expected the action to be omitted from the subject chain & the query itself be re-executed.